rust/datafusion/src/execution/physical_plan/expressions.rs

```diff
@@ -413,20 +413,7 @@ impl Max {
 
 impl AggregateExpr for Max {
     fn data_type(&self, input_schema: &Schema) -> Result<DataType> {
-        match self.expr.data_type(input_schema)? {
```

Comment:
I agree that the type of `Max` and `Min` should be the same as the input (as those functions are never going to overflow the value type of their input, the way `Sum` or `Avg` could. 

I actually think the change results in cleaner code too, which is a nice bonus